rcp/org.testeditor.rcp4.uatests/src/test/java/org/testeditor/rcp4/writetests/AutocompletionProposalTest.tcl

```diff
@@ -0,0 +1,140 @@
+import org.testeditor.rcp4.*
+import org.testeditor.fixture.swt.*
+
+# AutocompletionProposalTest
+
+config TestEditorConfig
+
+// the appropriate aml is added within the setup of this test
+
+* Check that "myApp" ^is proposed ^as component
+
+	Component: ActiveEditor
+	// given
+	- Kill content completely // remove any default generated file content	
+	- Type "# App\n\n* do something\n\nComponent: " into active editor
+	// when
+	- hasMyApp = Has autocompletion proposal "myApp" when entering ""
+	// then
+	- assert hasMyApp
+
+* Given an empty component reference _ content completion will already provide useful proposals _ "with wrong" ^is filtered since no element to satisfy the parameter ^is available
+
+	Component: ActiveEditor
+	// given
+	// when
+	- Use auto completion with "" and select "myApp"
+	- Type "\n" into active editor
+	// then
+	- hasDoSomethingInteraction = Has autocompletion proposal "do something \"count: long\" - test step" when entering ""
+	- hasDoDifferentlyInteraction = Has autocompletion proposal "do differently \"count: long\" - test step" when entering ""
+	- shouldHaveDoWithRight = Has autocompletion proposal "with right <element> - test step" when entering ""
+	- shouldNotHaveDoWithWrong = Has autocompletion proposal "with wrong <element> - test step" when entering ""
+	- assert hasDoSomethingInteraction
+	- assert hasDoDifferentlyInteraction
+	- assert shouldHaveDoWithRight
+	- assert !shouldNotHaveDoWithWrong
+
+* Given a prefix of "- " _ selection of content completion works and text inserted ^does ^not add superfluous dash
+
+	Component: ActiveEditor
+	// given
+	- Type "- " into active editor
+	// when
+	- Use auto completion with "" and select "do something \"count: long\" - test step"
+	- Type "\n" into active editor
+	// then
+	- singleDashedInteraction = Contains editor "\n- do something \"1\""
+	- assert singleDashedInteraction
+
+* Given a prefixed of "- do some" _ assist provides only the ones limited to the ones matching that prefix
+
+	Component: ActiveEditor
+	// given
+	- Type "\n- do some" into active editor
+	// when
+	- hasDoSome = Has autocompletion proposal "do something \"count: long\" - test step" when entering ""
+	- hasDoSomeDiff = Has autocompletion proposal "do something different \"count: long\" - test step" when entering ""
+	- shouldNotHaveDoDiff = Has autocompletion proposal "do differently \"count: long\" - test step" when entering ""
```

Comment:
the name "shouldNotHaveDoDiff" is a bit confusing. If I understand it correctly, the condition actually checks for the inverse of what the name implies. The assert below checks the inverse, too, but contains a "double negative" because of this.